File: DSA/Mathematics/Exactly 3 Divisors.cpp
class Solution{
    public:
    int exactly3Divisors(int n)
    {
        //Your code here
        int ans=0;
        for(int i=2;i*i<=n;i++){
            bool isPrime=true;
            for(int j=2;j*j<=i;j++){
                if(i%j==0){
                    isPrime=false;
                    break;
                }
            }
            if(isPrime){
                ans++;
            }
        }
        return ans;
    }
};

File: GeeksForGeeks-Problems/DSA/Mathematics/Exactly 3 Divisors.java
class Solution {
    public boolean checkPrime(int n){
        if(n<2) return false;
        if(n==2 || n==3) return true;
        for(int i=2;i*i<=n;i++){
            if(n%i==0) return false;
        }
        return true;
    }
    public int exactly3Divisors(int n) {
        // code here
        int ans=0;
        for(int i=0;i*i<=n;i++){
            if(checkPrime(i)) ans++;
        }
        return ans;
    }
}
